superset/databases/commands/create.py

```diff
@@ -46,6 +48,34 @@ def run(self) -> Model:
         try:
             # Test connection before starting create transaction
             TestConnectionDatabaseCommand(self._properties).run()
+        except DatabaseTestConnectionFailedError as ex:
+            event_logger.log_with_context(
+                action=f"db_creation_failed.{ex.__class__.__name__}",
+                engine=self._properties.get("sqlalchemy_uri", "").split(":")[0],
+            )
+            # if the exception is DatabaseTestConnectionFailedError it has errors in it
+            # so we try to return the message from the errors instead of using the
+            # default message
+            error_list: List[SupersetError] = ex.errors
+            for error in error_list:
+                # We could have just set the message coming from the original exception
+                # But that would have impacted all other places that relies on the
+                # current message. Like api_tests for postgres etc. In the future,
+                # after we discuss whether we want to show the original message
+                # all the time we can simply remove the check and set the message
+                # for all cases.
+                # Why Adding the message for this error type?
+                # Because it's the type returned for Service Accounts without roles
+                # or permissions when connecting to BigQuery DBs, which is
+                # the change we are introducing at this moment.
+                if (
+                    error.error_type
+                    == SupersetErrorType.CONNECTION_DATABASE_PERMISSIONS_ERROR
+                ):
+                    raise DatabaseConnectionFailedError(
```

Comment:
can we just raise the original `DatabaseTestConnectionFailedError`

Reply:
Updated. I am catching the new exception type in the API now along with sending back to the user a dict with the error messages instead of just raw strings.

Reply:
can you show me the diff between the payload coming back from the API when you send `DatabaseConnectionFailedError` vs the original exception?

Reply:
Before I was raising `DatabaseConnectionFailedError` which only has one error in it, so the API response would look like:
`{message: '<message_here>'}` while now I'm just throwing the original exception (`DatabaseTestConnectionFailed`) which has a list of errors, so the API would respond with a dict like: `{[error_type]: [error_message], [error_type_2]: [error_message_2], ...}` which is already a known pattern to the `DatabaseModal` for error displaying.

So basically after the change based on the input you gave me, I'm doing the latter.

For example:

Before when throwing a `DatabaseConnectionFailedError` the API would respond: